File: CoherenceCommons/java/src/com/seovic/coherence/util/processor/RemoveProcessor.java
package com.seovic.coherence.util.processor;


import com.tangosol.io.pof.PortableObject;
import com.tangosol.io.pof.PofReader;
import com.tangosol.io.pof.PofWriter;
import com.tangosol.util.processor.AbstractProcessor;
import com.tangosol.util.InvocableMap;
import java.io.IOException;


/**
 * @author Aleksandar Seovic  2009.06.24
 */
public class RemoveProcessor
        extends    AbstractProcessor
        implements PortableObject {

    private boolean m_fReturnOldValue;

    public RemoveProcessor() {
    }

    public RemoveProcessor(boolean fReturnOldValue) {
        m_fReturnOldValue = fReturnOldValue;
    }

    public Object process(InvocableMap.Entry entry) {
        Object oldValue = m_fReturnOldValue ? entry.getValue() : null;
        entry.remove(false);
        return oldValue;
    }
    public void readExternal(PofReader reader) throws IOException {
        m_fReturnOldValue = reader.readBoolean(0);
    }

    public void writeExternal(PofWriter writer) throws IOException {
        writer.writeBoolean(0, m_fReturnOldValue);
    }
}

File: main.net/src/Coherence.Tools/Coherence/Util/Processor/RemoveProcessor.cs
using System;
using Tangosol.IO.Pof;
using Tangosol.Net.Cache;
using Tangosol.Util.Processor;

namespace Seovic.Coherence.Util.Processor
{
    public class RemoveProcessor : AbstractProcessor, IPortableObject
    {
        private bool m_fReturnOldValue;

        public RemoveProcessor()
        {
        }

        public RemoveProcessor(bool fReturnOldValue)
        {
            m_fReturnOldValue = fReturnOldValue;
        }

        #region Overrides of AbstractProcessor

        public override object Process(IInvocableCacheEntry entry)
        {
            Object oldValue = m_fReturnOldValue ? entry.Value : null;
            entry.Remove(false);
            return oldValue;
        }

        #endregion

        #region Implementation of IPortableObject

        public void ReadExternal(IPofReader reader)
        {
            m_fReturnOldValue = reader.ReadBoolean(0);
        }

        public void WriteExternal(IPofWriter writer)
        {
            writer.WriteBoolean(0, m_fReturnOldValue);
        }

        #endregion
    }
}